src/main/java/com/sonymobile/jenkins/plugins/mq/mqnotifier/QueueListenerImpl.java

```diff
@@ -61,6 +62,10 @@ public void onLeft(Queue.LeftItem li) {
             json.put(Util.KEY_DEQUEUE_REASON, Util.VALUE_CANCELLED);
         } else {
             json.put(Util.KEY_DEQUEUE_REASON, Util.VALUE_BUILDING);
+            Label assignedLabel = li.getAssignedLabel();
+            json.put(Util.KEY_DEQUEUE_ALLOCATED_LABEL,
+                assignedLabel != null ? assignedLabel.getDisplayName() : Util.KEY_DEQUEUE_NO_LABEL);
```

Comment:
I though a bit about it yesterday, and went with NO_LABEL since I wanted an easy way to to ask kibana the question: "What builds does not have a label". Of course you can check the existing of fields, but that would make it harder to build certain kind of data like a histogram "labels vs queue times" and have one column for builds with not tied to a label.